code/hud/hudartillery.cpp

```diff
@@ -69,26 +69,35 @@ void parse_ssm(const char *filename)
 		while(required_string_either("#end", "$SSM:")) {
 			required_string("$SSM:");
 			ssm_info s;
+			int string_index;
 
 			// name
 			stuff_string(s.name, F_NAME, NAME_LENGTH);
+			if (*s.name == 0) {
+				sprintf(s.name, "SSM " SIZE_T_ARG, Ssm_info.size());
+				Warning(LOCATION, "SSM entries must have a name!  Assigning \"%s\"\n", s.name);
```

Comment:
Is there a particular reason to make this a `Warning` instead of just a piece of information `mprintf`'d to the debug log?